packages/upload-api/src/space-allocate.js

```diff
@@ -1,29 +1,19 @@
-import * as API from '../api.js'
+import * as API from './types.js'
 import * as Server from '@ucanto/server'
 import * as Space from '@web3-storage/capabilities/space'
 
 /**
- * @typedef {object} Context
- * @property {object} models
- * @property {API.SpaceProviderRegistry} models.provisions
- * @property {API.SpaceProviderRegistry} [models.spaces]
  *
  * @param {{capability: {with: API.SpaceDID, nb:{size:number}}}} input
- * @param {Context} context
+ * @param {API.SpaceServiceContext} context
  * @returns {Promise<API.Result<{size:number}, API.AllocationError>>}
  */
 export const allocate = async ({ capability }, context) => {
   const { with: space, nb } = capability
   const { size } = nb
-  const { provisions, spaces } = context.models
-  const result = await provisions.hasStorageProvider(space)
+  const result = await context.provisionsStorage.hasStorageProvider(space)
```

Comment:
Again not necessarily something this PR should solve, but we should instead be checking if space has provision for the `provider: invocation.audience.did()`. Or alternatively it could be `consumer/has` invocation from the service DID.

Reply:
yep agreed - I'd like to keep this out of this PR, but agree that we should move in this direction